cg/meta/workflow/downsample/downsample.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from pathlib import Path
+
+from cg.apps.slurm.slurm_api import SlurmAPI
+from cg.meta.workflow.downsample.sbatch import WORKFLOW_TEMPLATE
+from cg.models.cg_config import CGConfig
+from cg.store.models import Sample
+
+LOG = logging.getLogger("logger")
+
+
+class DownsampleWorkflow:
+    def __init__(
+        self,
+        config: CGConfig,
+        number_of_reads: int,
+        downsampled_sample: Sample,
+        dry_run: bool,
+        output_fastq_dir: str,
+        input_fastq_dir: str,
+        original_sample: Sample,
+    ):
+        if dry_run:
+            downsampled_sample = Sample()
+            original_sample = Sample()
+
+        self.config: CGConfig = config
+        self.downsampled_sample: Sample = downsampled_sample
+        self.number_of_reads: int = number_of_reads
+        self.dry_run: bool = dry_run
+        self.input_fastq_dir: str = input_fastq_dir
+        self.original_sample: Sample = original_sample
+        self.output_fastq_dir: str = output_fastq_dir
+        self.slurm_api: SlurmAPI = SlurmAPI()
+        self._account: str = "production"
+        self._memory: str = "100"
+        self._number_tasks: str = "2"
+        self._time: str = "10:00:00"
+        self._quality_of_service: str = "normal"
+        self._email: str = "hiseq.clinical@scilifelab.se"
+
+    @property
+    def account(self) -> str:
+        return self._account
+
+    @property
+    def email(self) -> str:
+        return self._email
+
+    @property
+    def bin_downsample(self) -> str:
+        """The path to the downsample.sh script on Hasta."""
+        return Path("home", "proj", "production", "bin", "downsample.sh").name
```

Comment:
I think this will only return the name of the script, probably want to extract it to a config as well